### PR TITLE
Fixes hdtp to pass native strings to JS

### PR DESCRIPTION
**Pull request by glebm**
hdtp API still accepts Racket strings so that doesn't change, but internally it now has to pass native strings to JS APIs.

### CODE REVIEW

**Comment by vishesh** [comment]
Thanks a lot for this! Did you try some examples from here https://github.com/vishesh/racketscript-playground/tree/master/examples ?

**Comment by glebm** [comment]
I've sent a PR to the playground as well. The only one I've tried and fixed for now is colored-carpet (only the jQuery calls needed fixing).

File: racketscript-extras/racketscript/htdp/image.rkt [hunk 1]
```diff
@@ -166,11 +167,11 @@
    (λ (ctx x y)
      (with-origin ctx [x y]
        (set-object! ctx
-                    [font          #js*.this.font]
-                    [textAlign     "center"]
-                    [textBaseline  "middle"]
-                    [fillStyle     #js*.this.color])
-       (#js.ctx.fillText #js*.this.text 0 0)))])
+                    [font          (js-string #js*.this.font)]
```
Comment (vishesh) [inline racketscript-extras/racketscript/htdp/image.rkt]: Can we convert this (and other js-string in render) when we are updating the object?

**Comment by vishesh** [comment]
Missed sending this:

@glebm Can you please do a quick check if games are running fine? Its a quick indicator if things are running fine. If something doesn't work, its a bug with HTDP or RacketScript. Since you fixed Playground, you can just run other games from there. Changes looks good, and hopefully nothing should fail! :)

File: racketscript-extras/racketscript/htdp/image.rkt (hunk 1 above)
Comment (glebm) [reply racketscript-extras/racketscript/htdp/image.rkt]: While we can, I think it is simpler Racket Strings internally and JS strings only when passing to JS APIs. This way Racket code can be used to manipulate strings, and it is a clear decision whether something is a Racket string or a JS string.

**Comment by glebm** [comment]
Yeah there are some issues with games, hopefully easy to fix, I'll look into it

**Comment by glebm** [comment]
@vishesh OK, done, should be good to merge.

All the games work as before. Flappy bird still has the accidental initialization issue but it is present on http://rapture.twistedplane.com:8080/#example/flappy-birds as well

**Comment by vishesh** [comment]
Thanks!

**Pull request merged**